### DISCUSSION TITLE
Wiring up NavLink active prop to react-router

### DISCUSSION THREAD

**Discussion by lukepighetti** [mantinedev/mantine#5788] [Q&A]
```typescript
<NavLink
  component={Link}
  active={???}
  to={`/app`}
  label="Links"
/>
```

Been poking around through the docs on both Mantine 7 and react-router-dom, and unfortunately haven't seen any clear path to connecting the `NavLink.active` prop to react-router-dom such that it is active after the `to` prop has been navigated to.

Any insights?

**Comment by rtivital** [comment]
Use https://reactrouter.com/en/main/utils/match-path

**Reply by corydeppen** [reply]
Do you think it might be possible to support the `[aria-current=page]` selector in the [`NavLink` styles](https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/NavLink/NavLink.module.css#L27)? Since Remix and React Router [set the `aria-current` attribute](https://remix.run/docs/en/main/components/nav-link#aria-current), it seems like this might allow their `NavLink` to be used as the root component for the Mantine `NavLink` without having to explicitly set the `active` prop.

**Reply by corydeppen** [reply]
@rtivital I was wondering if you'd be opposed to the approach mentioned above for including the `aria-current` selector or if you have other ideas for enabling folks using Remix, React Router, and others to use the (nav) link component provided by the framework inside `renderRoot` of a Mantine `NavLink`. In the case of Remix/React Router, the `NavLink` component already knows whether the link is for the active route so it wouldn't be ideal to duplicate that logic. Additionally, as [this article](https://amanexplains.com/using-aria-current-for-nav-links/) mentions, it's easy enough for Next.js apps to use the same `aria-current` attribute with their `Link` component.

**Reply by rtivital** [reply]
You can add any additional attributes to the root element by supplying them as props.

**Reply by corydeppen** [reply]
It's not about adding attributes in the case of Remix/React Router, as the `aria-current` attribute is already added by the framework/library. The issue is that Mantine only applies active styles for the `data-active` attribute. If Mantine expanded the [active style](https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/NavLink/NavLink.module.css#L27) to also include `aria-current`, I thought that might allow the active style to be applied when frameworks like Remix used `aria-current="page"` to indicate the link is for the active page.

**Reply by corydeppen** [reply]
I changed that one line in the generated styles and it appears to apply the active style when a Remix `NavLink` is used inside `renderRoot` without having to add anything to either the Mantine or Remix `NavLink` components.

```diff
- &:where([data-active]) {
+ &:where([data-active], [aria-current='page']) {
```

I'd be happy to submit a PR if it's something you might consider.

**Reply by rtivital** [reply]
Okay, I'm fine with it, you are welcome to submit a PR.

**Reply by corydeppen** [reply]
I've created #6180.
